### PR TITLE
epic(esl-event-listner): `ESLWheelTarget` proxy target to handle long wheel user actions

### LINKED ISSUE #2017: [🚀esl-event-listener]: request for InertWheel event detector
As a consumer, I want to be able to detect "inert" (long wheel) scrolls through the wheel control of the mouse (or alternative, e.g. long swipe on the touchpad).

During the consumer code research, the following snippet was found:
```ts

  @listen('wheel')
  protected _onWheel(event: WheelEvent): void {
    this._onWheelAggregated(event);
  }

  protected _onWheelAggregated = aggregate((events: WheelEvent[]) => {
    const dist = events.reduce((agg, e) => agg + e.deltaY, 0);
    if (dist > WHEEL_TOLERANCE) // Long wheel detected
  }, 100);

```

Proposed to simplify user flow in one of the following ways:

Base (event name, target and options are discussable)
```ts
 @listen({
    event: 'longwheel', 
    target: ($el: Element) => ESLWheelTarget.for($el, {timeout: 100, distance: 400})
  })
  protected _onWheel(event: WheelEvent): void {
     // Long wheel detected
  }
```

Open questions: 
  - Units for distance ... Not sure it is unified in browsers or even in specs.

### PR DESCRIPTION
**Pull request by fshovchko**
Closes #2017

### CODE REVIEW

**Comment by ala-n** [comment]
✅ Visual testing and review passed from my side

PS: But @exadel-inc/esl-core-maintainers, if you double-check me, it would be wonderful ⚡

**Review by ala-n — APPROVED** [review]

File: src/modules/esl-event-listener/README.md [hunk 1]
```diff
@@ -632,6 +632,43 @@ ESLEventUtils.subscribe(host, {
...
+ESLWheelTarget.for(el: Element, settings?: ESLWheelTargetSetting): ESLWheelTarget;
+```
+
+**Parameters**:
+
+- `el` - `Element` to listen for long wheel events on.
+- `settings` - optional settings (`ESLWheelTargetSetting`)
+
+The `ESLWheelTargetSetting` configuration includes these optional attributes:
+- `distance` - The minimum distance to accept as a long scroll in pixels (400 by default)
```
Comment (dshovchko) [inline src/modules/esl-event-listener/README.md:652]:
```suggestion
- `distance` - the minimum distance to accept as a long scroll in pixels (400 by default)
```

File: src/modules/esl-event-listener/README.md [hunk 2]
```diff
@@ -632,6 +632,43 @@ ESLEventUtils.subscribe(host, {
...
+- `timeout` - The maximum duration of the wheel events to consider it inertial in miliseconds (100 by default)
```
Comment (dshovchko) [inline src/modules/esl-event-listener/README.md:653]:
```suggestion
- `timeout` - the maximum duration of the wheel events to consider it inertial in milliseconds (100 by default)
```

**Review by dshovchko — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Review by HenadzV — APPROVED** [review]

File: src/modules/esl-event-listener/test/targets/wheel.test.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
+import {ESLWheelEvent, ESLWheelTarget} from '../../core/targets/wheel.target';
+
+describe('ESLWheelTarget', () => {
+  jest.useFakeTimers();
+
+  describe('ESLWheelTarget do not throws error on incorrect input (silent processing)', () => {
```
Comment (yadamskaya) [inline src/modules/esl-event-listener/test/targets/wheel.test.ts:6]:
```suggestion
  describe('ESLWheelTarget does not throw error on incorrect input (silent processing)', () => {
```

**Review by yadamskaya — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Pull request merged**